x-pack/filebeat/input/config.go

```diff
@@ -0,0 +1,11 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package input
+
+// Config has options common to all inputs' configurations
+type Config struct {
+	// KeepNull determines whether published events will keep null values or omit them.
+	KeepNull bool `config:"keep_null"`
+}
```

Comment:
Why using this common struct only in x-pack inputs? Will this option be available in OSS inputs? (I think it should be)

Reply:
It will be available in OSS inputs too. That is implemented here: https://github.com/elastic/beats/pull/13928/files#diff-b6244e3a701e1fad9969756499d1e8fcR49.

But I think maybe what you're suggesting is to move this struct into OSS code and embed it into both OSS and X-Pack code?

Reply:
I don't think we need to configure KeepNull per input here. The 'pipeline' being passed (inputs call ConnectWith) is an interface. Filebeat overwrites that interface in `filebeat/channel/connector.go`. Your PR alreay adapts the `pipelineConnector`, and therefore should set `KeepNull` for any input used with filebeat.